operations/mimir/autoscaling.libsonnet

```diff
@@ -283,6 +287,62 @@
     if !$._config.autoscaling_distributor_enabled then {} else removeReplicasFromSpec
   ),
 
+  // Cloud gateways
+
+  local newCloudGwScaledObject(name) = self.newScaledObject(
+    name, $._config.namespace, {
+      min_replica_count: $._config.autoscaling_cortex_gateway_min_replicas,
+      max_replica_count: $._config.autoscaling_cortex_gateway_max_replicas,
+      // Be aware that the default value for the trigger "metricType" field is "AverageValue"
+      // (see https://keda.sh/docs/2.9/concepts/scaling-deployments/#triggers), which means that KEDA will
+      // determine the target number of replicas by dividing the metric value by the threshold. This means in practice
+      // that we can sum together the values for the different replicas in our queries.
+      triggers: [
+        {
+          metric_name: '%s_cpu_hpa_%s' % [std.strReplace(name, '-', '_'), $._config.namespace],
+
+          // To scale out relatively quickly, but scale in slower, we look at the average CPU utilization per cloud gateway over 5m (rolling window)
+          // and then we pick the highest value over the last 15m.
+          // Multiply by 1000 to get the result in millicores. This is due to KEDA only working with ints.
+          query: 'max_over_time(sum(rate(container_cpu_usage_seconds_total{container="%s",namespace="%s"}[5m]))[15m:]) * 1000' % [
+            name,
+            $._config.namespace,
+          ],
+          // Threshold is expected to be a string
+          threshold: std.toString(cpuToMilliCPUInt($.cortex_gateway_container_options.resources.requests.cpu)),
+        },
+        {
+          metric_name: '%s_memory_hpa_%s' % [std.strReplace(name, '-', '_'), $._config.namespace],
+
+          // To scale out relatively quickly, but scale in slower, we look at the max memory utilization across all cloud gateways over 15m.
+          query: 'max_over_time(sum(container_memory_working_set_bytes{container="%s",namespace="%s"})[15m:])' % [
+            name,
+            $._config.namespace,
+          ],
+
+          // Threshold is expected to be a string
+          threshold: std.toString(siToBytes($.cortex_gateway_container_options.resources.requests.memory)),
+        },
+      ],
+    },
+  ),
+
+  cortex_gateway_scaled_object: if !$._config.autoscaling_cortex_gateway_enabled then null else newCloudGwScaledObject(
```

Comment:
"cortex gateway" is not a thing in the OSS jsonnet. I don't we should have it here. I'm fine keeping the changes to the mixin.

Reply:
@pracucci ah yes, that's true.